mlrun/package/packagers_manager.py

```diff
@@ -0,0 +1,698 @@
+
+        # Go over the artifacts and link:
+        for artifact in self.artifacts:
+            # Go over the extra data keys:
+            for key in artifact.spec.extra_data:
+                # Future link is marked with ellipses (...):
+                if artifact.spec.extra_data[key] is ...:
+                    # Look for an artifact or result with this key to link it:
+                    extra_data = self._look_for_extra_data(
+                        key=key, artifacts=joined_artifacts, results=joined_results
+                    )
+                    # Print a warning if a link is missing:
+                    if extra_data is None:
+                        logger.warn(
+                            f"Could not find {key} to link as extra data for {artifact.key}."
+                        )
+                    # Link it (None will be used in case it was not found):
+                    artifact.spec.extra_data[key] = extra_data
+            # Go over the metrics keys if available (`ModelArtifactSpec` has a metrics property that may be waiting for
+            # values from logged results):
+            if hasattr(artifact.spec, "metrics"):
+                for key in artifact.spec.metrics:
+                    # Future link is marked with ellipses (...):
+                    if artifact.spec.metrics[key] is ...:
+                        # Link it (None will be used in case it was not found):
+                        artifact.spec.metrics[key] = joined_results.get(key, None)
+
+    def clear_packagers_outputs(self):
+        """
+        Clear the outputs of all packagers. This method should be called at the end of the run after logging all
+        artifacts as some will require uploading the files that will be deleted in this method.
+        """
+        for packager in self._packagers:
+            for path in packager.get_clearing_path_list():
+                if not os.path.exists(path):
+                    continue
+                if os.path.isdir(path):
+                    shutil.rmtree(path)
+                else:
+                    os.remove(path)
+
+    class _InstructionsNotesKey:
+        """
+        Library of keys for the packager instructions to be added to the packed artifact's spec.
+        """
+
+        PACKAGER_NAME = "packager_name"
+        OBJECT_TYPE = "object_type"
+        ARTIFACT_TYPE = "artifact_type"
+        INSTRUCTIONS = "instructions"
+
+    def _try_to_collect_packagers_from_module(
+        self, module_name: str, packagers: List[str]
+    ):
+        """
+        Collect a packagers of a given module only if it was successfully imported.
+
+        :param module_name: The module name to try to import.
+        :param packagers:   The packagers to collect.
+        """
+        try:
+            importlib.import_module(module_name)
+
+            self.collect_packagers(packagers=packagers)
+        except ModuleNotFoundError:
+            pass
+
+    def _collect_packagers(self):
```

Comment:
```suggestion
    def _collect_mlrun_packagers(self):
```

Reply:
I want to move the default packagers to collect to the context handler, so the manager will be cleaner. Empty unless told to collect packagers. WDYT?